services/alertmanager/registry_test.go

```diff
@@ -53,6 +55,7 @@ func TestRegistry(t *testing.T) {
 
 		expected := &ammodels.PostableAlert{
 			Annotations: annotations,
+			EndsAt:      strfmt.DateTime(nowValueCopy.Add(resolveTimeoutFactor * resendInterval)),
```

Comment:
Add method does not modify the receiver, so there is no need to use `nowValueCopy`

Reply:
I tried `nowValue` but the test failed because the `ext` field in `time.Time` didn't match so I made a copy

Reply:
```
--- FAIL: TestRegistry (0.00s)
    --- FAIL: TestRegistry/DelayFor (0.00s)
        registry_test.go:68: 
            	Error Trace:	registry_test.go:68
            	Error:      	Not equal: 
            	            	expected: &ammodels.PostableAlert{Annotations:ammodels.LabelSet{"annotation":"test"}, EndsAt:strfmt.DateTime{wall:0xbfb03dcac15cd572, ext:151570362490, loc:(*time.Location)(0x11c76e0)}, StartsAt:strfmt.DateTime{wall:0x0, ext:0, loc:(*time.Location)(nil)}, Alert:ammodels.Alert{GeneratorURL:"", Labels:ammodels.LabelSet{"label":"demo"}}}
            	            	actual  : &ammodels.PostableAlert{Annotations:ammodels.LabelSet{"annotation":"test"}, EndsAt:strfmt.DateTime{wall:0xbfb03dbbc15cd572, ext:91570362490, loc:(*time.Location)(0x11c76e0)}, StartsAt:strfmt.DateTime{wall:0x0, ext:0, loc:(*time.Location)(nil)}, Alert:ammodels.Alert{GeneratorURL:"", Labels:ammodels.LabelSet{"label":"demo"}}}
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -4,3 +4,3 @@
            	            	  },
            	            	- EndsAt: (strfmt.DateTime) 2020-06-10T06:37:31.022Z,
            	            	+ EndsAt: (strfmt.DateTime) 2020-06-10T06:36:31.022Z,
            	            	  StartsAt: (strfmt.DateTime) 0001-01-01T00:00:00.000Z,
            	Test:       	TestRegistry/DelayFor
FAIL

```

Reply:
Well, that's because we modify nowValue just below :) I think we should change that

Reply:
but if we don't modify the `now` value below then we won't get an alert from the `r.colect()` call

Reply:
so maybe we should extract value of `endsAt` to a variable before we change `now`?